### PR TITLE
change socket‘s permission to 0777

### PR DESCRIPTION
**Pull request by lelemka0**
#554 #342

### CODE REVIEW

**Comment by AkinoKaede** [comment]
I don't think it is a good method.

**Comment by lelemka0** [comment]
> I don't think it is a good method.

https://github.com/golang/go/issues/11822#issuecomment-123850227
This seems to be the one of recommended methods.
Another method (syscall.Umask) will affect all files.

**Comment by ljxfstorm** [comment]
As from nginx/src/core/ngx_connection.c:
```
            if (ls[i].sockaddr->sa_family == AF_UNIX) {
                mode_t   mode;
                u_char  *name;

                name = ls[i].addr_text.data + sizeof("unix:") - 1;
                mode = (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH);

                if (chmod((char *) name, mode) == -1) {
                    ngx_log_error(NGX_LOG_EMERG, cycle->log, ngx_errno,
                                  "chmod() \"%s\" failed", name);
                }

                if (ngx_test_config) {
                    if (ngx_delete_file(name) == NGX_FILE_ERROR) {
                        ngx_log_error(NGX_LOG_EMERG, cycle->log, ngx_errno,
                                      ngx_delete_file_n " %s failed", name);
                    }
                }
            }
```
Mode 0666 may be a better choice.

File: transport/internet/system_listener.go [hunk 1]
```diff
@@ -80,6 +83,11 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
 		policyFunc := func(upstream net.Addr) (proxyproto.Policy, error) { return proxyproto.REQUIRE, nil }
 		l = &proxyproto.Listener{Listener: l, Policy: policyFunc}
 	}
+	if err == nil && ds {
+		if e := os.Chmod(address, 0666); e != nil {
```
Comment (rurirei) [inline transport/internet/system_listener.go:87]: move this to the `ds body (if body)` above. there's no need to check nil error and create a temp bool.

Comment (lelemka0) [reply transport/internet/system_listener.go]: ok

Comment (rurirei) [reply transport/internet/system_listener.go]: do not create unnecessary bool `ds = true`, and handle err in the domain_socket body.
```golang
... ...
		} else {
			// normal unix domain socket needs lock
			locker := &FileLocker{
				path: address + ".lock",
			}
			if err := locker.Acquire(); err != nil {
				return nil, err
			}
			if err := os.Chmod(address, 0666); err != nil {
				return nil, err
			}
			ctx = context.WithValue(ctx, address, locker)
		}
```

**Comment by hmol233** [comment]
感谢各位的支持与贡献。
正如 @AkinoKaede 所说，这并不是一个好方法。
稍后我会另外实现一个。

**Pull request closed without merging**